sessiontxn/isolation/readcommitted.go

```diff
@@ -251,7 +343,12 @@ func (p *PessimisticRCTxnContextProvider) AdviseOptimizeWithPlan(val interface{}
 		plan = execute.Plan
 	}
 
-	if v, ok := plan.(*plannercore.Insert); ok && v.SelectPlan == nil {
+	useLastOracleTS := planSkipGetTsoFromPD(p.sctx, plan, false)
+
+	if useLastOracleTS {
+		failpoint.Inject("tsoUseConstantFuture", func() {
+			sessiontxn.TsoUseConstantCountInc(p.sctx)
+		})
```

Comment:
Should the failpoint be added to the `ConstantFuture.Wait` instead so it's more general?

Reply:
good idea

Reply:
Oh, but TsoUseConstantCountInc need a parameter 'session', `ConstantFuture.Wait` can't get the param.